src/libprojplugin/libprojplugin.cpp

```diff
@@ -104,13 +146,75 @@ QString LibprojPlugin::readProjectFile()
 bool LibprojPlugin::parseMetadata(const QString & strJson)
 {
     if (strJson.isNull() || strJson.isEmpty()) {
-        qWarning() << "Nothing to parse.";
+        qWarning() << "Nothing to parse";
         return false;
     }
     else {
-        bool ok = false;
-        parsedMetadata = QtJson::parse(strJson, ok).toMap();
-        Q_ASSERT(ok);
-        return true;
+        std::string err;
+        qDebug() << "About to start parsing";
+        projectData = Json::parse(strJson.toStdString(), err);
+        if (!err.empty())
+            qWarning() << "Something wrong with parsing:\n" + QString(err.c_str());
+        return err.empty() ? true : false;
     }
 }
+
+void LibprojPlugin::triggerAddNewFileAction()
+{
+   switch(QMessageBox::question(nullptr, tr("File exists?"),tr("Are you want to add existing file (Yes) or create new (No)?"),
+                                QMessageBox::Yes, QMessageBox::No))
+   {
+   case QMessageBox::StandardButton::Yes:
+   {
+       /* TODO
+        * ...
+        */
+       break;
+   }
+   case QMessageBox::StandardButton::No:
+   {
+       QString newFilename (QFileDialog::getSaveFileName(nullptr, tr("Where you want to save new file?"), project->document()->filePath().toFileInfo().dir().path()));
+       QFile newFile (newFilename);
+       if (!newFile.exists())
+           newFile.open(QIODevice::ReadWrite | QIODevice::Text);
+       else {
+           qWarning() << "File already exists!";
```

Comment:
What do you mean about splitting? Must I write function for
opening/creating files and executing it here (under cases)?
19 ÌÀÔ. 2015 11:14, "Maxim Kot" <notifications@github.com> ÎÁÐÉÓ.

> In src/libprojplugin/libprojplugin.cpp
> <https://github.com/drewpts/LPROJ-main/pull/20#discussion_r24973744>:
>
> > +   {
> > +   case QMessageBox::StandardButton::Yes:
> > +   {
> > +       /* TODO
> > +        * ...
> > +        */
> > +       break;
> > +   }
> > +   case QMessageBox::StandardButton::No:
> > +   {
> > +       QString newFilename (QFileDialog::getSaveFileName(nullptr, tr("Where you want to save new file?"), project->document()->filePath().toFileInfo().dir().path()));
> > +       QFile newFile (newFilename);
> > +       if (!newFile.exists())
> > +           newFile.open(QIODevice::ReadWrite | QIODevice::Text);
> > +       else {
> > +           qWarning() << "File already exists!";
>
> What about return error code, or just false for example? You don't split
> gui and logis so you have such problems
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/drewpts/LPROJ-main/pull/20/files#r24973744>.
>